### PR TITLE
Crypto key refactor

### LINKED ISSUE #212: Crypro Error "not enough data"
Here and there this error happens:

```
Error: error:0D06B08E:asn1 encoding routines:asn1_d2i_read_bio:not enough data
    at Sign.sign (node:internal/crypto/sig:131:29)
    at CryptoNode.signPkcs8 (/Users/ingof/Dev/GitHub/matter.js/packages/matter-node.js/dist/crypto/CryptoNode.js:115:44)
    at Function.Crypto.signPkcs8 (/Users/ingof/Dev/GitHub/matter.js/packages/matter.js/dist/cjs/crypto/Crypto.js:46:68)
    at Function.daCertToAsn1 (/Users/ingof/Dev/GitHub/matter.js/packages/matter.js/dist/cjs/certificate/CertificateManager.js:211:75)
    at AttestationCertificateManager.generateDaCert (/Users/ingof/Dev/GitHub/matter.js/packages/matter.js/dist/cjs/certificate/AttestationCertificateManager.js:143:59)
    at AttestationCertificateManager.getDACert (/Users/ingof/Dev/GitHub/matter.js/packages/matter.js/dist/cjs/certificate/AttestationCertificateManager.js:48:23)
    at new CommissioningServer (/Users/ingof/Dev/GitHub/matter.js/packages/matter.js/dist/cjs/CommissioningServer.js:100:54)
    at Device.<anonymous> (/Users/ingof/Dev/GitHub/matter.js/packages/matter-node.js-examples/dist/examples/DeviceNode.js:112:41)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/ingof/Dev/GitHub/matter.js/packages/matter-node.js-examples/dist/examples/DeviceNode.js:11:58) {
  library: 'asn1 encoding routines',
  function: 'asn1_d2i_read_bio',
  reason: 'not enough data',
  code: 'ERR_OSSL_ASN1_NOT_ENOUGH_DATA'
}
```

We should find out how to work areound that or at least catch-and-retry

### PR DESCRIPTION
**Pull request by lauckhart**
Replace key handling with a JS implementation that converts keys to standard JWK Key format.  Streamlines the Crypto.ts API and removes the various formats and hacks we previously used to pass keys to Node's OpenSSL wrapper.

fixes #212
could be considered phase 1 of #122

### CODE REVIEW

File: packages/matter-node.js/test/cluster/ClusterServerTestingUtil.ts [hunk 1]
```diff
@@ -12,6 +12,12 @@ import { FabricIndex, VendorId, FabricId, NodeId } from "@project-chip/matter.js
 import { ByteArray } from "@project-chip/matter.js/util";
 import { Attributes, ClusterServerObj, Commands, Events } from "@project-chip/matter.js/cluster";
 import { Endpoint } from "@project-chip/matter.js/device";
+import { PrivateKey } from "../../src/crypto/export";
```
Comment (Apollon77) [inline packages/matter-node.js/test/cluster/ClusterServerTestingUtil.ts:15]: Fix import please @project-chip/matter.js/crypto

**Review by Apollon77 — APPROVED** [review]
Please fix the two imports, then great!

I verified with apple and google. backward compatibility is also given

File: packages/matter-node.js/test/cluster/ClusterServerTestingUtil.ts (hunk 1 above)
Comment (lauckhart) [reply packages/matter-node.js/test/cluster/ClusterServerTestingUtil.ts]: done

File: packages/matter-node.js/test/session/secure/CasePairingTest.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ import * as assert from "assert";
 import { TlvEncryptedDataSigma2, TlvEncryptedDataSigma3, TlvSignedData } from "@project-chip/matter.js/session";
 import { ByteArray } from "@project-chip/matter.js/util";
 import { CryptoNode } from "../../../src/crypto/CryptoNode";
+import { PublicKey } from "../../../src/crypto/export";
```
[the remark this answers is not in the document]
Comment (lauckhart) [reply packages/matter-node.js/test/session/secure/CasePairingTest.ts]: done

File: packages/matter-node.js/test/crypto/CryptoTest.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ import * as assert from "assert";
 import { CryptoNode } from "../../src/crypto/CryptoNode";
```
Comment (Apollon77) [inline packages/matter-node.js/test/crypto/CryptoTest.ts:8]: please this one too, and please check if more

File: packages/matter-node.js/test/session/secure/CasePairingTest.ts [hunk 4]
```diff
@@ -7,8 +7,8 @@
 import * as assert from "assert";
 import { TlvEncryptedDataSigma2, TlvEncryptedDataSigma3, TlvSignedData } from "@project-chip/matter.js/session";
 import { ByteArray } from "@project-chip/matter.js/util";
+import { PublicKey } from "@project-chip/matter.js/crypto";
 import { CryptoNode } from "../../../src/crypto/CryptoNode";
```
Comment (Apollon77) [inline packages/matter-node.js/test/session/secure/CasePairingTest.ts:11]: please this one too, and please check if more

File: packages/matter-node.js/test/crypto/CryptoTest.ts (hunk 3 above)
Comment (Apollon77) [reply packages/matter-node.js/test/crypto/CryptoTest.ts:8]: bullshit

**Pull request merged**